File: Modules/HumanResource/TrainingManagement/ProgramSession/ProgramSessionColumns.cs
using Serenity.ComponentModel;
using System;
using System.Collections.Generic;
using System.ComponentModel;

namespace HRMSoftware.TrainingManagement.Columns;

[ColumnsScript("TrainingManagement.ProgramSession")]
[BasedOnRow(typeof(ProgramSessionRow), CheckNames = true)]
public class ProgramSessionColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string ProgramName { get; set; }
    [Width(200), ProgramDepartmentListFormatter]
    public List<int> DepartmentList { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public int Status { get; set; }
    // public int InsertUserId { get; set; }
    // public DateTime InsertDate { get; set; }
    // public int UpdateUserId { get; set; }
    // public DateTime UpdateDate { get; set; }
    // public int DeleteUserId { get; set; }
    // public DateTime DeleteDate { get; set; }
}

File: Modules/ServerTypes/TrainingManagement/ProgramSessionColumns.ts
import { ProgramDepartmentListFormatter } from "@/HumanResource/TrainingManagement/ProgramDepartment/ProgramDepartmentListFormatter";
import { ProgramSessionStatusType } from "./ProgramSessionStatusType";

export class ProgramSessionColumns {
    static columnsKey = 'TrainingManagement.ProgramSession';
}

[ProgramDepartmentListFormatter, ProgramSessionStatusType]; // referenced types